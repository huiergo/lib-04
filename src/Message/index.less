@import (reference) '../style/variables.less';

@prefix: '__lib-04';
@c-primary: 'green';

.__lib-04__message-text {
  background-color: red;
    // font-size:  68 * @rpx;
    //   background-color:@c-primary;
    //   [data-prefers-color=dark] & {
    //       background-color: #ccc;
    //     }
    //     [data-prefers-color=orange] & {
    //       background-color: orange;
    //     }
    //     [data-prefers-color=green] & {
    //       background-color: green;
    //     }
  }