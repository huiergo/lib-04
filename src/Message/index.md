---
nav:
  title: 组件
  path: /components
group:
  title: components
---

## Message

<code src="./demo/basic.tsx"></code>

### 基本用法:

```tsx
import React from 'react';
import { Message } from 'lib-04';

export default () => (
  <Message kind="warning" theme="green">
    这是一条警示
  </Message>
);
```

<API src="./index.tsx"></API>

More skills for writing demo: https://d.umijs.org/guide/basic#write-component-demo
