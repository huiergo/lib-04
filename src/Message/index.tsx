// 源文件
import React,{PropsWithChildren} from 'react';
import t from 'prop-types'
import { usePrefersColor} from 'dumi/theme'
import './index.less'

export interface MessageProps{
   /**
    * @description Message 的类型
    * @default 'info'
    */
    kind?:'info'|'positive'|'negative'|'warning'
     /**
    * @description 主题色
    * @default 'blue'
    */
   theme?:string
}

export type KindMap = Record<Required<MessageProps>['kind'],string>

const kinds:KindMap={
    info: "red",
    positive: "green",
    negative: "blue",
    warning: "purple"
}

const Message: React.FC<PropsWithChildren<MessageProps>>=({children, kind="info",...rest})=>{
    const [color, setColor]=usePrefersColor()
    return (
        <div
            className='__lib-04__message'
            style={{color:rest.theme}}
            {...rest}
        >
            {children}
            <div  className='__lib-04__message-text'>自定义的主题色配置为：{rest.theme}</div>
            <div>Global theme color {color}</div>
        </div>
    )
}

Message.propTypes={
    kind:t.oneOf(['info','positive','negative','warning'])
}

export default Message