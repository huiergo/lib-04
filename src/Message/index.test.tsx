import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Message from './index';

describe('<Message />', () => {
  it('render Message with dumi', () => {
    const msg = 'dumi';

    render(<Message kind='info'/>);
    expect(screen.queryByText(msg)).toBeInTheDocument();
  });

  test('should render default',()=>{
    const {container} =render(<Message>default</Message>)
    expect(container).toMatchSnapshot()
  })
  test('should render alert with type',()=>{
    const kinds: any[] = ['info', 'warning', 'positive', 'negative'];
    const { getByText } = render(
      <>
        {kinds.map((k) => (
          <Message kind={k} key={k}>
            {k}
          </Message>
        ))}
      </>,
    );
    kinds.forEach((k) => {
      expect(getByText(k)).toMatchSnapshot();
    });
  })
});
