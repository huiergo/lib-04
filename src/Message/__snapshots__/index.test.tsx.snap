// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render alert with type 1`] = `
<div
  class="__lib-04__message"
>
  info
  <div
    class="__lib-04__message-text"
  >
    自定义的主题色配置为：
  </div>
  <div>
    Global theme color 
    red
  </div>
</div>
`;

exports[`<Message /> should render alert with type 2`] = `
<div
  class="__lib-04__message"
>
  warning
  <div
    class="__lib-04__message-text"
  >
    自定义的主题色配置为：
  </div>
  <div>
    Global theme color 
    red
  </div>
</div>
`;

exports[`<Message /> should render alert with type 3`] = `
<div
  class="__lib-04__message"
>
  positive
  <div
    class="__lib-04__message-text"
  >
    自定义的主题色配置为：
  </div>
  <div>
    Global theme color 
    red
  </div>
</div>
`;

exports[`<Message /> should render alert with type 4`] = `
<div
  class="__lib-04__message"
>
  negative
  <div
    class="__lib-04__message-text"
  >
    自定义的主题色配置为：
  </div>
  <div>
    Global theme color 
    red
  </div>
</div>
`;

exports[`<Message /> should render default 1`] = `
<div>
  <div
    class="__lib-04__message"
  >
    default
    <div
      class="__lib-04__message-text"
    >
      自定义的主题色配置为：
    </div>
    <div>
      Global theme color 
      red
    </div>
  </div>
</div>
`;
